backend/api/Areas/Admin/Controllers/AdministrativeAreaController.cs

```diff
@@ -0,0 +1,180 @@
+        [ProducesResponseType(typeof(Model.AdministrativeAreaModel), 200)]
+        [ProducesResponseType(typeof(Api.Models.ErrorResponseModel), 400)]
+        [SwaggerOperation(Tags = new[] {"admin-administrative-area"})]
+        public IActionResult UpdateAdministrativeAreaAsync([FromBody] Model.AdministrativeAreaModel model)
+        {
+            var entity = _mapper.Map<Entity.AdministrativeArea>(model);
+            _pimsAdminService.AdministrativeArea.Update(entity);
+
+            var adminArea = _mapper.Map<Model.AdministrativeAreaModel>(entity);
+            return new JsonResult(adminArea);
+        }
+
+        /// <summary>
+        /// POST - Add a new administrative area to the datasource.
+        /// </summary>
+        /// <param name="model">The agency model.</param>
+        /// <returns>The administrative area added.</returns>
+        [HttpPost]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(Model.AdministrativeAreaModel), 201)]
+        [ProducesResponseType(typeof(Api.Models.ErrorResponseModel), 400)]
+        [SwaggerOperation(Tags = new[] {"admin-administrative-area"})]
+        public IActionResult AddAdministrativeArea([FromBody] Model.AdministrativeAreaModel model)
+        {
+            var entity = _mapper.Map<Entity.AdministrativeArea>(model);
+            _pimsAdminService.AdministrativeArea.Add(entity);
+
+            var administrativeArea = _mapper.Map<Model.AdministrativeAreaModel>(entity);
+
+            return CreatedAtAction(nameof(GetAdministrativeArea), new {id = administrativeArea.Id}, administrativeArea);
+        }
+
+        /// <summary>
+        /// DELETE - Delete the administrative area from the datasource.
+        /// </summary>
+        /// <param name="model">The administrative area model.</param>
+        /// <returns>The administrative area that was deleted.</returns>
+        [HttpDelete("{id}")]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(Model.AdministrativeAreaModel), 200)]
+        [ProducesResponseType(typeof(Api.Models.ErrorResponseModel), 400)]
+        [SwaggerOperation(Tags = new[] {"admin-administrative-area"})]
+        public IActionResult DeleteAdministrativeAreaAsync([FromBody] Model.AdministrativeAreaModel model)
+        {
+            var entity = _mapper.Map<Entity.AdministrativeArea>(model);
+            _pimsAdminService.AdministrativeArea.Remove(entity);
+
+            return new JsonResult(model);
+        }
+
+        /// <summary>
+        /// Check if name of admin area is available
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet("check/name-available")]
+        [HasPermission(Permissions.SystemAdmin)]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(Model.CheckAreaAvailabilityModel), 200)]
+        [SwaggerOperation(Tags = new[] { "admin-administrative-area" })]
+        public IActionResult IsAdministrativeAreaAvailable(string name, int id)
```

Comment:
Don't really understand why the `id` is also included in this endpoint?

Reply:
I was wondering if this was necessary but for example if you were updating an already existing area and it ran the check for duplication, would it say this name isn't available since it would be comparing against itself at one point? Or am I overthinking this one? I guess it is not that big of a deal either since there is one field to edit at the moment

Reply:
hmmm, you have a point.  You're essentially asking if the name exists but isn't the current administrative area you are editing.  Hmmm, it's a bit of an ugly implementation.  I would actually recommend you don't do this validation at all.  Just make sure the table has a unique constraint on the name.  Then if they try it the DB will throw an error.  You can then inform the user of the unique requirement failure after they attempt to save it.

I also suspect that you will need to update the state with a new list of administrative areas any time the user add/delete/update them.

You could also check the state that contains all the current administrative areas to see if the name has already been used.  If you want a quick user experience.